### PR TITLE
Improve error management

### PR DESCRIPTION
**Pull request by sujeebant**
- Check if the url list passed to the constructor is valid
- Forward socket error events

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -142,22 +160,16 @@ SignalhubWs.prototype.close = function (cb) {
...
+  this.once('close:channels', () => {
+    this.emit('close')
+  })
+}
 
+SignalhubWs.prototype._closeChannels = function () {
   const len = this.channels.size
   if (len === 0) {
-    this.emit('close')
+    this.emit('close:channels')
```
Comment (soyuka) [inline index.js]: why not `close` directly? It's forwarding the event to close?

File: index.js [hunk 2]
```diff
@@ -167,7 +179,7 @@ SignalhubWs.prototype._closeChannels = function (cb) {
       channel.end(() => {
         if (++closed === len) {
           this.channels.clear()
-          this.emit('close')
+          this.emit('close:channels')
```
Comment (soyuka) [inline index.js]: why not `close` directly? It's forwarding the event to close?

File: index.js [hunk 3]
```diff
@@ -142,22 +160,16 @@ SignalhubWs.prototype.close = function (cb) {
...
-    return
-  }
-  this.closed = true
-
-  if (cb) {
-    this.on('close', cb)
-  }
+  this.once('close:channels', () => {
+    this.emit('close')
+  })
```
Comment (soyuka) [inline index.js]: not really useful see below

File: index.js (hunk 1 above)
Comment (sujeebant) [reply index.js]: We can close directly, I'll fix it.

**Comment by sujeebant** [comment]
The "close:channels" event has been removed and "close" is sent directly from the _closeChannels function

**Comment by soyuka** [comment]
thanks @sujeebant

**Pull request merged**